File: Chapter 03/Python/section_3_4_order_processor_refactor/before/order.py
from dataclasses import dataclass
from typing import List

"""
A Simple Data Transfer Object (DTO).
* ARCHITECTURE NOTE: This is a perfect use of a Python dataclass. 
It is a passive data container. The problem we will solve in the refactor 
isn't this data, but the way the Processor class "reaches into" it 
to perform disparate business rules.
"""
@dataclass
class Order:
    items: List[str]
    total: float
    customer_email: str

File: Chapter 03/JS/section_3_4_order_processor_refactor/before/order.js
/**
 * A Simple Data Transfer Object (DTO).
 * * ARCHITECTURE NOTE: This class is a pure data container. The problem 
 * isn't the data itself, but rather that the logic for every single 
 * part of the business process is currently being "shovelled" into 
 * a single processor class.
 */
class Order {
    constructor(items, total, customerEmail) {
        this.items = items;
        this.total = total;
        this.customerEmail = customerEmail;
    }
}

module.exports = Order;